### PR TITLE
Convert bug report to new syntax

### LINKED ISSUE #4944: Issue Forms
GitHub is rolling out [Issue Forms](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms) - a more formal way of creating interactive forms during issue submission. A number of our existing templates would be better implemented as Forms, when someone has the time/interest.

### PR DESCRIPTION
**Pull request by paulbalandan**
**Description**
Closes #4944 

**Checklist:**
- [x] Securely signed commits

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+body:
+  - type: markdown
+    attributes:
+      value: |
+        Thanks for taking the time to fill out this bug report! Before you begin, please
+        ensure that there are no existing issues, whether still open or closed, related
+        to your report. If there is, your report will be closed promptly.
+
+  - type: input
```
Comment (MGatner) [inline .github/ISSUE_TEMPLATE/bug_report.yml:14]: Should this be a dropdown? There are only a few options right?

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    id: codeigniter-version
+    attributes:
+      label: 'CodeIgniter4 Version'
+    validations:
+      required: true
+
+  - type: checkboxes
+    id: operating-systems
+    attributes:
+      label: Which operating systems have you used?
```
Comment (MGatner) [inline .github/ISSUE_TEMPLATE/bug_report.yml:31]:
```suggestion
      label: Which operating systems have you tested for the bug?
```

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        - label: macOS
+        - label: Windows
+        - label: Linux
+    validations:
+      required: true
+
+  - type: dropdown
+    id: server
+    attributes:
+      label: Which SAPI did you use?
```
Comment (MGatner) [inline .github/ISSUE_TEMPLATE/bug_report.yml:43]: Will people understand SAPI? Would "web server software" or something be more clear?

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+  - type: textarea
+    attributes:
+      label: Steps to Reproduce
+      description: Steps to reproduce the behavior.
+    validations:
+      required: true
+
+  - type: textarea
+    attributes:
+      label: 'Expected Output'
```
Comment (MGatner) [inline .github/ISSUE_TEMPLATE/bug_report.yml:79]: Quoting on labels is inconsistent. I'm not sure if it matters.

**Review by MGatner — CHANGES_REQUESTED** [review]
This is awesome! Thanks for tackling this - it's been on my mind for a while but just never sat down with the file format. A few comments...

**Comment by sfadschm** [comment]
Why is there two descriptions asking for the expected behavior?
Lines 73 and 88.

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+    attributes:
+      label: Database
+    validations:
+      required: false
+
+  - type: textarea
+    id: description
+    attributes:
+      label: What happened?
+      description: Also tell us, what did you expect to happen?
```
Comment (kenjis) [inline .github/ISSUE_TEMPLATE/bug_report.yml:73]: This should ask only what happened. 

Not to be expected to happen.
It is asked in `Expected Output`.

**Review by MGatner — COMMENTED** [review]
I agree with Kenji's change, then this is ready IMO.

**Review by kenjis — APPROVED** [review]

**Comment by paulbalandan** [comment]
I'll be pushing a direct commit. There's a typo

```
There is a problem with this template

body[1]: dropdowm is not a valid input type. Learn more.
```

**Comment by kenjis** [comment]
Where did you get the error message?
There is no new Issue form, I was looking for why.

**Comment by paulbalandan** [comment]
In the file itself, there is a validation message.

**Comment by paulbalandan** [comment]
The new issue form looks savvy. 🎉

**Pull request merged**